### PR TITLE
fix(masthead): added menusection null checks

### PR DESCRIPTION
**Pull request by IgnacioBecerra**
### Related Ticket(s)
#8322

### Description
When the`menuSection` object is not found, the React masthead will fail to render and will cause an error that might crash the website.

### Changelog

**New**

- more null checks for the `menuSection` object

### CODE REVIEW

File: packages/react/src/components/Masthead/Masthead.js [hunk 1]
```diff
@@ -256,7 +256,7 @@ const Masthead = ({
       if (!matchFound) {
         if (sections.url === currentUrlPath) {
           matchFound = true;
-        } else if (sections?.menuSections[0]) {
+        } else if (sections?.menuSections && sections?.menuSections[0]) {
```
Comment (emyarod) [inline packages/react/src/components/Masthead/Masthead.js:259]: I believe we can use optional chaining here too right?

```suggestion
        } else if (sections?.menuSections?.[0]) {
```

File: packages/react/src/components/Masthead/MastheadLeftNav.js [hunk 2]
```diff
@@ -124,14 +124,18 @@ const MastheadLeftNav = ({
         rest.hasL1Data ? 'l1' : 'l0'
       }-nav${i}`;
 
-      const menuItems = link.menuSections[0]?.menuItems;
+      const menuItems = link.menuSections && link.menuSections[0]?.menuItems;
```
Comment (emyarod) [inline packages/react/src/components/Masthead/MastheadLeftNav.js:127]:
```suggestion
      const menuItems = link.menuSections?.[0]?.menuItems;
```

File: packages/react/src/components/Masthead/MastheadLeftNav.js [hunk 3]
```diff
@@ -269,10 +273,11 @@ function _renderLevel1Submenus(
     let highlightedItems = [];
     const items = [];
 
-    menu.sections[0].menuItems.forEach(item => {
-      if (item.highlighted) return highlightedItems.push(item);
-      return items.push(item);
-    });
+    menu.sections &&
+      menu.sections[0].menuItems.forEach(item => {
```
Comment (emyarod) [inline packages/react/src/components/Masthead/MastheadLeftNav.js:277]:
```suggestion
    menu.sections?.[0].menuItems.forEach(item => {
```

File: packages/react/src/components/Masthead/MastheadLeftNav.js [hunk 4]
```diff
@@ -286,7 +291,7 @@ function _renderLevel1Submenus(
             menuState.level0 === menu.parentKey && menuState.level1 >= 0,
         })}
         id={`panel__(${menu.parentKey},-1)`}
-        heading={menu.sections[0]?.heading}
+        heading={menu.sections && menu.sections[0]?.heading}
```
Comment (emyarod) [inline packages/react/src/components/Masthead/MastheadLeftNav.js:294]:
```suggestion
        heading={menu.sections?.[0]?.heading}
```

**Review by emyarod — APPROVED** [review]
looks good to me! if I'm not mistaken I think we can use optional chaining for some more of these checks

**Comment by IgnacioBecerra** [comment]
@emyarod Weird, that's actually the first thing I tried and it had given me a different error. Could be that I was still missing other null checks at the time, but regardless the last commit now uses optional chaining. Thanks!

**Pull request merged**